packages/core/src/components/discord-embed/DiscordEmbed.ts

```diff
@@ -0,0 +1,402 @@
+			border-radius: 4px;
+			contain: paint;
+			display: block;
+			grid-column: 1/1;
+			margin-top: 16px;
+		}
+
+		.discord-embed .discord-embed-media.discord-embed-media-video {
+			height: 225px;
+		}
+
+		.discord-embed .discord-embed.media .discord-embed-image {
+			overflow: hidden;
+			position: relative;
+			user-select: text;
+		}
+
+		.discord-embed .discord-embed-media .discord-embed-video {
+			-webkit-box-align: center;
+			-webkit-box-pack: center;
+			align-items: center;
+			border-radius: 0;
+			cursor: pointer;
+			display: flex;
+			height: 100%;
+			justify-content: center;
+			max-height: 100%;
+			width: 100%;
+
+			width: 400px;
+			height: 225px;
+			left: 0px;
+			top: 0px;
+		}
+
+		.discord-embed-custom-emoji {
+			display: inline-block;
+		}
+
+		.discord-embed-custom-emoji .discord-embed-custom-emoji-image {
+			width: 18px;
+			height: 18px;
+			vertical-align: bottom;
+		}
+
+		slot[name='footer']::slotted(*) {
+			grid-column: 1/3;
+			grid-row: auto/auto;
+		}
+	`;
+
+	/**
+	 * The color to use for the embed's left border.
+	 * Can be any [CSS color value](https://www.w3schools.com/cssref/css_colors_legal.asp).
+	 */
+	@property()
+	public color: string;
+
+	/**
+	 * The author's name.
+	 */
+	@property({ attribute: 'author-name' })
+	public authorName: string;
+
+	/**
+	 * The author's avatar URL.
+	 */
+	@property({ attribute: 'author-image' })
+	public authorImage: string;
+
+	/**
+	 * The URL to open when you click on the author's name.
+	 */
+	@property({ attribute: 'author-url' })
+	public authorUrl: string;
+
+	/**
+	 * The embed title.
+	 */
+	@property({ attribute: 'embed-title' })
+	public embedTitle: string;
+
+	/**
+	 * The URL to open when you click on the embed title.
+	 */
+	@property()
+	public url: string;
+
+	/**
+	 * The thumbnail image to use.
+	 */
+	@property()
+	public thumbnail: string;
+
+	/**
+	 * The embed image to use (displayed at the bottom).
+	 */
+	@property()
+	public image: string;
+
+	/**
+	 * The embed video to use (displayed at the bottom, same slot as the image).
+	 * @important YouTube videos will not be playable on your projects, this is due to YouTube using DASH to play their videos rather
+	 * than providing the raw media stream (in a container such as mp4 or ogg). Links to regular MP4 files (such as on a CDN) however
+	 * will autoplay!
+	 * @note Video takes priority over image.
+	 * @remark Providing both a video and an image will ensure the image is shown to users with browsers
+	 * that do not support HTML5 video playback.
+	 * @example https://download.blender.org/peach/bigbuckbunny_movies/big_buck_bunny_1080p_stereo.ogg
+	 */
+	@property()
+	public video: string;
+
+	/**
+	 * The provider to show above the embed, for example for YouTube videos it will show "YouTube" at the top of the embed (above the author)
+	 * @example YouTube
+	 */
+	@property()
+	public provider: string;
+
+	// TODO: private hasPerformedRerenderChecks: 'dirty' | 'pristine' = 'pristine';
```

Comment:
Putting a comment on this line so we don't forget. Ref: https://github.com/TheBv/discord-components/blob/8320ddfac3cc3390fe08b3f7a7c151eeed3861bd/packages/core/src/components/discord-embed/discord-embed.tsx#L91-L100